src/platform/packages/private/kbn-reporting/get_csv_panel_actions/panel_actions/get_csv_panel_action.tsx

```diff
@@ -166,6 +191,7 @@ export class ReportingCsvPanelAction implements ActionDefinition<EmbeddableApiCo
    * @deprecated
    */
   private executeDownload = async (params: ExecutionParams) => {
+    // Deprecated and does not work for ES|QL mode
```

Comment:
The conflict was in how to handle the Immediate Download action. Since ES|QL is not supported for this action, I left it as it. The default behaviour without `xpack.reporting.csv.enablePanelActionDownload: true` is fixed with this backport.